### DISCUSSION TITLE
DANGER, inserting XXXX at the end of handedOutList

### DISCUSSION THREAD

**Discussion by tomweston78** [javers/javers#1228] [Q&A]
Hi, 

We started seeing these errors in the logs recently, they seem have no visible effect on the data processing overall but I'd like to ask if anyone faced those too and what could be a potential effect and mitigation?

```
ERROR 29454 --- [or-http-epoll-4] org.javers.core.commit.HandedOutIds      : DANGER, inserting 131554 at the end of handedOutList: 
ERROR 29454 --- [or-http-epoll-4] org.javers.core.commit.HandedOutIds      : [131637.00, 131636.00, 131635.00, 131634.00, 131633.00, 131632.00, ... ]

```
We use mostly javers.commit() where javers is a standard instance from SpringBoot starter, nothing fancy. JaVers version: 6.7.0

**Comment by bartoszwalacik** [comment]
It's hard to say what is the reason. The HandedOutIds class implements a cache, but it's isn't tested enough for concurrent access. Its API is simple and it can be unit tested. I would start from writing a good unit test with high level of concurrency.

**Reply by peter-krk** [reply]
I've also noticed such error when using  javers.commit() - any clarifications here?

**Reply by bartoszwalacik** [reply]
nothing new, if you guys suspect a bug -- start from writing a good unit test with high level of concurrency
